rrfs-test/IODA/python/bufr2ioda_nexrad.py

```diff
@@ -0,0 +1,358 @@
+
+import sys
+import argparse
+import numpy as np
+import numpy.ma as ma
+import calendar
+import json
+import time
+import copy
+import math
+import datetime
+import os
+from datetime import datetime
+from pyioda import ioda_obs_space as ioda_ospace
+from wxflow import Logger
+from pyiodaconv import bufr
+from collections import namedtuple
+import warnings
+# suppress warnings
+warnings.filterwarnings('ignore')
+
+
+def Mask_typ_for_var(typ, var):
+
+    typ_var = copy.deepcopy(typ)
+    for i in range(len(typ_var)):
+        if ma.is_masked(var[i]):
+            typ_var[i] = typ.fill_value
+
+    return typ_var
+
+
+def bufr_to_ioda(config, logger):
+
+    subsets = config["subsets"]
+    logger.debug(f"Checking subsets = {subsets}")
+
+    # Get parameters from configuration
+    data_format = config["data_format"]
+    data_type = config["data_type"]
+    data_description = config["data_description"]
+    data_provider = config["data_provider"]
+    cycle_type = config["cycle_type"]
+    dump_dir = config["dump_directory"]
+    ioda_dir = config["ioda_directory"]
+    cycle = config["cycle_datetime"]
+
+    # Get derived parameters
+    yyyymmdd = cycle[0:8]
+    hh = cycle[8:10]
+    reference_time = datetime.strptime(cycle, "%Y%m%d%H")
+    reference_time = reference_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # General informaton
+    converter = 'BUFR to IODA Converter'
+    platform_description = 'NEXRAD'
+
+    logger.info(f"reference_time = {reference_time}")
+
+    bufrfile = f"{cycle_type}.t{hh}z.{data_type}.tm00.{data_format}"
+    DATA_PATH = os.path.join(dump_dir, bufrfile)
+    if not os.path.isfile(DATA_PATH):
+        logger.info(f"DATA_PATH {DATA_PATH} does not exist")
+        return
+    logger.debug(f"The DATA_PATH is: {DATA_PATH}")
+
+    # ============================================
+    # Make the QuerySet for all the data we want
+    # ============================================
+    start_time = time.time()
+
+    logger.info('Making QuerySet')
+    q = bufr.QuerySet(subsets)
+
+    # ObsType
+    #q.add('observationType', '*/TYP')
+
+    # MetaData
+    q.add('year', '*/YEAR')
+    q.add('month', '*/MNTH')
+    q.add('day', '*/DAYS')
+    q.add('hour', '*/HOUR')
+    q.add('minute', '*/MINU')
+    q.add('second', '*/SECO')
+    q.add('latitude', '[*/CLATH, */CLAT]')
+    q.add('longitude', '[*/CLONH, */CLON]')
+    q.add('stationIdentification', '*/SSTN')
+    q.add('height', '*/HSMSL')
+    q.add('heightOfAntenna', '*/HSALG')
+
+    # ObsValue
+    q.add('beamAzimuthAngle', '*/ANAZ')
+    q.add('beamTiltAngle', '*/ANEL')
+    q.add('gateRange', '*/NL2RW{1}/DIST125M')
+    q.add('radialVelocity', '*/NL2RW{1}/DMVR')
+    q.add('spectralWidth', '*/NL2RW{1}/DVSW')
+    q.add('ppiVolume', '*/VOID')
+    q.add('ppiIndex', '*/SCID')
+    q.add('unfoldingVelocity', '*/HNQV')
+    q.add('volumeCoveragePattern', '*/VOCP')
+
+    # QualityMarker
+    q.add('windAlongRadialLineQM', '*/QCRW')
```

Comment:
Can we use radialWindQM for QCRW?

Reply:
@ShunLiu-NOAA done!

Reply:
It looks like QCRW was changed to `radialVelocityQM` instead of `radialWindQM`. Does this matter @ShunLiu-NOAA?